tests/test_Sequence.cc

```diff
@@ -3911,3 +3911,77 @@ TEST_CASE("Sequence: Check setter/getter function for step setup script", "[Sequ
         REQUIRE(ctx.step_setup_script == script);
     }
 }
+
+TEST_CASE("Sequence: sequence timeout", "[Sequence]")
+{
+    Step step{Step::type_action};
+    step.set_script("sleep(0.2)");
+
+    Sequence seq{"test_sequence"};
+    seq.push_back(std::move(step));
+
+    seq.set_timeout(100ms);
+    REQUIRE(seq.get_timeout() == 100ms);
```

Comment:
I think the difference between the timeout duration and the sleep duration may be a little too low. Especially on the slow Github CI runners execution can easily be interrupted for 100 ms. So I'd recommend a sleep of some seconds and, to speed the test up, a timeout of 10 ms or so.